src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java

```diff
@@ -88,6 +88,8 @@
   /** Error message to display on unknown communication type. */
   String comsErrMsg = "Unknown communication type given. Unable to build payload.";
 
+  public static int MAX_PAYLOAD_NUM_CHARACTERS = 0;
```

Comment:
The max is zero? Should this just be 
```suggestion
  public static int MAX_PAYLOAD_NUM_CHARACTERS;
```

Reply:
I set that to 0 because TagPayload uses it in it's constructor which is called after the  OSIsoft JSON payload is constructed and that gets set to a valid value but there were still warnings about it having the potential to be uninitialized in TagPayload constructor. Some better error handling could probably be added there

Reply:
I think better error handling is probably a good idea.

Reply:
Im never going to say no to better error handling :)